src/test/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/wiremock/NomisSyncApiMockServer.kt

```diff
@@ -0,0 +1,53 @@
+package uk.gov.justice.digital.hmpps.prisonerfromnomismigration.wiremock
+
+import com.fasterxml.jackson.databind.ObjectMapper
+import com.github.tomakehurst.wiremock.WireMockServer
+import com.github.tomakehurst.wiremock.client.WireMock.aResponse
+import com.github.tomakehurst.wiremock.client.WireMock.get
+import com.github.tomakehurst.wiremock.client.WireMock.urlMatching
+import org.junit.jupiter.api.extension.AfterAllCallback
+import org.junit.jupiter.api.extension.BeforeAllCallback
+import org.junit.jupiter.api.extension.BeforeEachCallback
+import org.junit.jupiter.api.extension.ExtensionContext
+import org.springframework.test.context.junit.jupiter.SpringExtension
+
+class NomisSyncApiExtension :
+  BeforeAllCallback,
+  AfterAllCallback,
+  BeforeEachCallback {
+  companion object {
+    @JvmField
+    val nomisSyncApi = NomisSyncApiMockServer()
+    lateinit var objectMapper: ObjectMapper
+  }
+
+  override fun beforeAll(context: ExtensionContext) {
+    nomisSyncApi.start()
+    objectMapper = (SpringExtension.getApplicationContext(context).getBean("jacksonObjectMapper") as ObjectMapper)
+  }
+
+  override fun beforeEach(context: ExtensionContext) {
+    nomisSyncApi.resetAll()
+  }
+
+  override fun afterAll(context: ExtensionContext) {
+    nomisSyncApi.stop()
+  }
+}
+
+class NomisSyncApiMockServer : WireMockServer(WIREMOCK_PORT) {
+  companion object {
+    private const val WIREMOCK_PORT = 8098
+  }
```

Comment:
Also needed to generalise the sync API mock server so that it can be used by different test suites

Reply:
Not sure I understand this; but lets have a chat tomorrow